Add Plugin: spotify-links

Cutaiar opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

https://github.com/Cutaiar/obsidian-spotify-links

## Release Checklist
- [x] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [x]  ~~Android~~ (desktop only)
  - [x]  ~~iOS~~ (desktop only)
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
- [x] I have given proper attribution to these other projects in my `README.md`.

joethei commented:
[const redirectUri = "obsidian://callback"; // Not sure if this is a reasonable redirect URI but this works](https://github.com/Cutaiar/obsidian-spotify-links/blob/35abe78fab7bb68eea327f8469c92c54bb5801c2/src/main.ts#L28)
You should make this more unique to your plugin, like `obsidian://spotify-callback` for example.

[const body = await fetch(url, payload);](https://github.com/Cutaiar/obsidian-spotify-links/blob/35abe78fab7bb68eea327f8469c92c54bb5801c2/src/spotifyAPI.ts#L64)
Use the `requestUrl` function from the Obsidian API instead, it will handle some things like CORS automatically.

Cutaiar commented:
> [const redirectUri = "obsidian://callback"; // Not sure if this is a reasonable redirect URI but this works](https://github.com/Cutaiar/obsidian-spotify-links/blob/35abe78fab7bb68eea327f8469c92c54bb5801c2/src/main.ts#L28) You should make this more unique to your plugin, like `obsidian://spotify-callback` for example.
> 
> [const body = await fetch(url, payload);](https://github.com/Cutaiar/obsidian-spotify-links/blob/35abe78fab7bb68eea327f8469c92c54bb5801c2/src/spotifyAPI.ts#L64) Use the `requestUrl` function from the Obsidian API instead, it will handle some things like CORS automatically.

Thanks for the feedback! I've addressed your comments and the changes have been merged into master. I plan to make a `v0.2.0` release which includes these fixes as well as some other general improvements. I will update here when that it released.

Cutaiar commented:
By the way, I have this video which demonstrates the main functionality.

https://www.youtube.com/watch?v=Qzs3ssqKk2Y&t=1s

Cutaiar commented:
Alright, version `0.2.0` is [released](https://github.com/Cutaiar/obsidian-spotify-links/releases/tag/0.2.0). It addresses the comments above as well as a few other things. I am confident this version is ready to go out to the public.
